<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Visual DB</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_3">
    <property name="margin">
     <number>3</number>
    </property>
    <property name="spacing">
     <number>3</number>
    </property>
    <item row="0" column="0">
     <widget class="Gui::TabWidget" name="mTabWidget">
      <property name="toolTip">
       <string>Tab widget</string>
      </property>
      <property name="whatsThis">
       <string>The tab widget is used for displaying graphics scene and description tabs.</string>
      </property>
      <property name="documentMode">
       <bool>false</bool>
      </property>
      <property name="tabsClosable">
       <bool>false</bool>
      </property>
      <property name="movable">
       <bool>false</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="mMenuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="mFileMenu">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <widget class="QMenu" name="mSessionMenu">
     <property name="title">
      <string>Sessions</string>
     </property>
     <addaction name="mLoadSessionAction"/>
     <addaction name="mSaveSessionAction"/>
    </widget>
    <addaction name="mNewConnectionAction"/>
    <addaction name="separator"/>
    <addaction name="mSaveToImgAction"/>
    <addaction name="separator"/>
    <addaction name="mShowOptionsDialogAction"/>
    <addaction name="mSessionMenu"/>
    <addaction name="separator"/>
    <addaction name="mShowPrintPreviewDialogAction"/>
    <addaction name="mShowPrintDialogAction"/>
    <addaction name="separator"/>
    <addaction name="mExitAction"/>
   </widget>
   <widget class="QMenu" name="mEditMenu">
    <property name="title">
     <string>Edit</string>
    </property>
   </widget>
   <widget class="QMenu" name="mViewMenu">
    <property name="title">
     <string>&amp;View</string>
    </property>
    <addaction name="mSetFullScreenAction"/>
    <addaction name="mShowTableListAction"/>
    <addaction name="mShowLogPanelAction"/>
   </widget>
   <widget class="QMenu" name="mSchemeMenu">
    <property name="title">
     <string>&amp;Scheme</string>
    </property>
    <addaction name="mSelectAllTablesAction"/>
    <addaction name="mRemoveAllTablesAction"/>
    <addaction name="mShowGridAction"/>
    <addaction name="mAlignToGridAction"/>
    <addaction name="mDivideIntoPagesAction"/>
    <addaction name="mShowLegendAction"/>
    <addaction name="mShowControlWidgetAction"/>
    <addaction name="mColorizeAccordingSchemasAction"/>
   </widget>
   <widget class="QMenu" name="mTableMenu">
    <property name="title">
     <string>&amp;Item</string>
    </property>
    <addaction name="mDeleteTableAction"/>
    <addaction name="mShowFieldsTypesAction"/>
    <addaction name="mHideFieldsTypesAction"/>
    <addaction name="mShowIndicesAction"/>
    <addaction name="mHideIndicesAction"/>
    <addaction name="mSetTableColorAction"/>
    <addaction name="mAdjustTableSizeAction"/>
    <addaction name="mGroupItemsAction"/>
    <addaction name="mUngroupItemsAction"/>
    <addaction name="mAnchorAction"/>
    <addaction name="mDisableAnchorAction"/>
    <addaction name="mSelectAllTablesInSchemaAction"/>
   </widget>
   <addaction name="mFileMenu"/>
   <addaction name="mEditMenu"/>
   <addaction name="mViewMenu"/>
   <addaction name="mSchemeMenu"/>
   <addaction name="mTableMenu"/>
  </widget>
  <widget class="QStatusBar" name="mStatusBar"/>
  <widget class="QToolBar" name="mToolBar">
   <property name="windowTitle">
    <string>Toolbar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="mNewConnectionAction"/>
   <addaction name="mSaveToImgAction"/>
   <addaction name="separator"/>
   <addaction name="mReloadDataAction"/>
   <addaction name="mDrawFullDBAction"/>
   <addaction name="separator"/>
   <addaction name="mRemoveAllTablesAction"/>
  </widget>
  <widget class="Gui::DockWidget" name="mLogPanelWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="MinimumExpanding" vsizetype="Minimum">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="maximumSize">
    <size>
     <width>524287</width>
     <height>150</height>
    </size>
   </property>
   <property name="baseSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="features">
    <set>QDockWidget::AllDockWidgetFeatures</set>
   </property>
   <property name="allowedAreas">
    <set>Qt::BottomDockWidgetArea|Qt::RightDockWidgetArea</set>
   </property>
   <property name="windowTitle">
    <string>Log panel</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="mDock">
    <layout class="QGridLayout" name="gridLayout_2">
     <property name="margin">
      <number>3</number>
     </property>
     <property name="spacing">
      <number>3</number>
     </property>
     <item row="0" column="0">
      <widget class="Gui::LogPanel" name="mLogPanel" native="true">
       <property name="toolTip">
        <string>Log panel</string>
       </property>
       <property name="whatsThis">
        <string>The log panel widget is used for displaying apps output.</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="Gui::DockWidget" name="mDatabaseTreeWidget">
   <property name="features">
    <set>QDockWidget::AllDockWidgetFeatures</set>
   </property>
   <property name="allowedAreas">
    <set>Qt::BottomDockWidgetArea|Qt::LeftDockWidgetArea</set>
   </property>
   <property name="windowTitle">
    <string>Database window</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QGridLayout" name="gridLayout">
     <property name="margin">
      <number>3</number>
     </property>
     <property name="spacing">
      <number>3</number>
     </property>
     <item row="0" column="0">
      <widget class="QTabWidget" name="mTreeTabWidget">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="tabPosition">
        <enum>QTabWidget::West</enum>
       </property>
       <property name="currentIndex">
        <number>-1</number>
       </property>
       <property name="tabsClosable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="mNewConnectionAction">
   <property name="icon">
    <iconset resource="../../visual_db.qrc">
     <normaloff>:/img/connect_creating.png</normaloff>:/img/connect_creating.png</iconset>
   </property>
   <property name="text">
    <string>New connection...</string>
   </property>
   <property name="statusTip">
    <string>Establishing of new connection</string>
   </property>
   <property name="whatsThis">
    <string/>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
   <property name="menuRole">
    <enum>QAction::TextHeuristicRole</enum>
   </property>
  </action>
  <action name="mSaveToImgAction">
   <property name="icon">
    <iconset resource="../../visual_db.qrc">
     <normaloff>:/img/filesave.png</normaloff>:/img/filesave.png</iconset>
   </property>
   <property name="text">
    <string>Save to image...</string>
   </property>
   <property name="statusTip">
    <string>Save scheme to image</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="mShowOptionsDialogAction">
   <property name="icon">
    <iconset resource="../../visual_db.qrc">
     <normaloff>:/img/configure.png</normaloff>:/img/configure.png</iconset>
   </property>
   <property name="text">
    <string>Options...</string>
   </property>
   <property name="statusTip">
    <string>Edit options</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="mLoadSessionAction">
   <property name="text">
    <string>Load session...</string>
   </property>
  </action>
  <action name="mSaveSessionAction">
   <property name="text">
    <string>Save session...</string>
   </property>
  </action>
  <action name="mShowPrintPreviewDialogAction">
   <property name="text">
    <string>Show preview...</string>
   </property>
  </action>
  <action name="mShowPrintDialogAction">
   <property name="text">
    <string>Print...</string>
   </property>
  </action>
  <action name="mExitAction">
   <property name="icon">
    <iconset resource="../../visual_db.qrc">
     <normaloff>:/img/exit.png</normaloff>:/img/exit.png</iconset>
   </property>
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="statusTip">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="mSetFullScreenAction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Full screen</string>
   </property>
   <property name="statusTip">
    <string>Full screen</string>
   </property>
   <property name="shortcut">
    <string>F11</string>
   </property>
  </action>
  <action name="mShowTableListAction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Database tree</string>
   </property>
  </action>
  <action name="mShowLogPanelAction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Log panel</string>
   </property>
  </action>
  <action name="mSelectAllTablesAction">
   <property name="text">
    <string>Select all</string>
   </property>
   <property name="statusTip">
    <string>Select all</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="mRemoveAllTablesAction">
   <property name="icon">
    <iconset resource="../../visual_db.qrc">
     <normaloff>:/img/eraser.png</normaloff>:/img/eraser.png</iconset>
   </property>
   <property name="text">
    <string>Remove all</string>
   </property>
   <property name="statusTip">
    <string>Remove all from scheme</string>
   </property>
  </action>
  <action name="mShowGridAction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show grid</string>
   </property>
   <property name="statusTip">
    <string>Show grid</string>
   </property>
  </action>
  <action name="mAlignToGridAction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Align to grid</string>
   </property>
   <property name="statusTip">
    <string>Align to the grid</string>
   </property>
  </action>
  <action name="mDivideIntoPagesAction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Divide into pages</string>
   </property>
   <property name="statusTip">
    <string>Divide into pages</string>
   </property>
  </action>
  <action name="mShowLegendAction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Show legend</string>
   </property>
   <property name="statusTip">
    <string>Show legend</string>
   </property>
  </action>
  <action name="mShowControlWidgetAction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show control widget</string>
   </property>
   <property name="statusTip">
    <string>Show control widget</string>
   </property>
  </action>
  <action name="mColorizeAccordingSchemasAction">
   <property name="text">
    <string>Colorize according schemas</string>
   </property>
   <property name="statusTip">
    <string>Colorize according schemas</string>
   </property>
  </action>
  <action name="mDeleteTableAction">
   <property name="icon">
    <iconset resource="../../visual_db.qrc">
     <normaloff>:/img/remove.png</normaloff>:/img/remove.png</iconset>
   </property>
   <property name="text">
    <string>Delete</string>
   </property>
   <property name="statusTip">
    <string>Delete item from the scheme</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
  </action>
  <action name="mShowFieldsTypesAction">
   <property name="icon">
    <iconset resource="../../visual_db.qrc">
     <normaloff>:/img/show.png</normaloff>:/img/show.png</iconset>
   </property>
   <property name="text">
    <string>Show fields' types</string>
   </property>
   <property name="statusTip">
    <string>Show fields' types</string>
   </property>
  </action>
  <action name="mHideFieldsTypesAction">
   <property name="icon">
    <iconset resource="../../visual_db.qrc">
     <normaloff>:/img/hide.png</normaloff>:/img/hide.png</iconset>
   </property>
   <property name="text">
    <string>Hide fields' types</string>
   </property>
   <property name="statusTip">
    <string>Hide fields' types</string>
   </property>
  </action>
  <action name="mShowIndicesAction">
   <property name="text">
    <string>Show indices</string>
   </property>
   <property name="toolTip">
    <string>Show indices</string>
   </property>
   <property name="statusTip">
    <string>Show indices</string>
   </property>
  </action>
  <action name="mHideIndicesAction">
   <property name="text">
    <string>Hide indices</string>
   </property>
   <property name="toolTip">
    <string>Hide indices</string>
   </property>
   <property name="statusTip">
    <string>Hide indices</string>
   </property>
  </action>
  <action name="mSetTableColorAction">
   <property name="icon">
    <iconset resource="../../visual_db.qrc">
     <normaloff>:/img/colors.png</normaloff>:/img/colors.png</iconset>
   </property>
   <property name="text">
    <string>Item color...</string>
   </property>
   <property name="statusTip">
    <string>Set color</string>
   </property>
  </action>
  <action name="mAdjustTableSizeAction">
   <property name="icon">
    <iconset resource="../../visual_db.qrc">
     <normaloff>:/img/adjustsize.png</normaloff>:/img/adjustsize.png</iconset>
   </property>
   <property name="text">
    <string>Adjust</string>
   </property>
   <property name="statusTip">
    <string>Adjust size</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+J</string>
   </property>
  </action>
  <action name="mGroupItemsAction">
   <property name="icon">
    <iconset resource="../../visual_db.qrc">
     <normaloff>:/img/group.png</normaloff>:/img/group.png</iconset>
   </property>
   <property name="text">
    <string>Group items</string>
   </property>
   <property name="statusTip">
    <string>Group items</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+G</string>
   </property>
  </action>
  <action name="mUngroupItemsAction">
   <property name="icon">
    <iconset resource="../../visual_db.qrc">
     <normaloff>:/img/ungroup.png</normaloff>:/img/ungroup.png</iconset>
   </property>
   <property name="text">
    <string>Ungroup items</string>
   </property>
   <property name="statusTip">
    <string>Ungroup items</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+U</string>
   </property>
  </action>
  <action name="mAnchorAction">
   <property name="icon">
    <iconset resource="../../visual_db.qrc">
     <normaloff>:/img/anchor.png</normaloff>:/img/anchor.png</iconset>
   </property>
   <property name="text">
    <string>Anchor</string>
   </property>
   <property name="statusTip">
    <string>Set anchor for selected</string>
   </property>
  </action>
  <action name="mDisableAnchorAction">
   <property name="text">
    <string>Disable anchor</string>
   </property>
   <property name="statusTip">
    <string>Disable anchor for selected</string>
   </property>
  </action>
  <action name="mSelectAllTablesInSchemaAction">
   <property name="text">
    <string>Select all in schema</string>
   </property>
  </action>
  <action name="mReloadDataAction">
   <property name="icon">
    <iconset resource="../../visual_db.qrc">
     <normaloff>:/img/reload.png</normaloff>:/img/reload.png</iconset>
   </property>
   <property name="text">
    <string>Refresh</string>
   </property>
   <property name="statusTip">
    <string>Reload</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="mDrawFullDBAction">
   <property name="icon">
    <iconset resource="../../visual_db.qrc">
     <normaloff>:/img/pencil.png</normaloff>:/img/pencil.png</iconset>
   </property>
   <property name="text">
    <string>Draw full DB</string>
   </property>
  </action>
  <action name="mAddTableAction">
   <property name="text">
    <string>Add</string>
   </property>
   <property name="statusTip">
    <string>Add</string>
   </property>
  </action>
  <action name="mDescribeObjectAction">
   <property name="text">
    <string>Describe</string>
   </property>
   <property name="statusTip">
    <string>Get object description</string>
   </property>
   <property name="shortcut">
    <string>Alt+Return</string>
   </property>
  </action>
  <action name="mQueryDataAction">
   <property name="text">
    <string>Query data</string>
   </property>
   <property name="statusTip">
    <string>Run SQL command for data retrieving</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Gui::TabWidget</class>
   <extends>QTabWidget</extends>
   <header location="global">gui/TabWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>Gui::DockWidget</class>
   <extends>QDockWidget</extends>
   <header location="global">gui/DockWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>Gui::LogPanel</class>
   <extends>QWidget</extends>
   <header location="global">gui/LogPanel.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../visual_db.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>mExitAction</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mShowTableListAction</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>showDatabaseTree(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mShowLogPanelAction</sender>
   <signal>toggled(bool)</signal>
   <receiver>mLogPanel</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>571</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mDrawFullDBAction</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>drawFullDbScheme()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mShowPrintPreviewDialogAction</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>showPrintPreviewDialog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mShowPrintDialogAction</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>showPrintDialog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mAddTableAction</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>addItem()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mDescribeObjectAction</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>describeObject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mQueryDataAction</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>queryData()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mSaveSessionAction</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveSession()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mLoadSessionAction</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>loadSession()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mSetFullScreenAction</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>setFullScreen(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>connectionDialogRequest(bool)</signal>
  <slot>drawFullDbScheme()</slot>
  <slot>showOptionsDialog()</slot>
  <slot>showPrintPreviewDialog()</slot>
  <slot>showPrintDialog()</slot>
  <slot>showDatabaseTree(bool)</slot>
  <slot>addItem()</slot>
  <slot>describeObject()</slot>
  <slot>queryData()</slot>
  <slot>saveSession()</slot>
  <slot>loadSession()</slot>
  <slot>loadLastSession()</slot>
  <slot>setFullScreen(bool)</slot>
  <slot>printPreview(QPrinter*)</slot>
  <slot>reloadData()</slot>
  <slot>addCommand(QUndoCommand*)</slot>
  <slot>showLogPanel(bool)</slot>
 </slots>
</ui>
